graph.go

```diff
@@ -95,7 +95,9 @@ type NodeRemover interface {
 // EdgeSetter is an interface for adding edges to a graph.
 type EdgeSetter interface {
 	// SetEdge adds an edge from one node to another.
-	// If the nodes do not exist, they are added.
+	// If the graph supports node addition the nodes
+	// will be added if they do not exist, otherwise
```

Comment:
This is a more general question/comment not related strictly to this PR but one that occurs to me frequently: I find the use of future tense in the documentation of interfaces slightly inappropriate. Unlike in the case of actual functions, here we don't know what the implementation will do. We want to say what is the intent of the method (e.g., adds an edge) or how it must behave (e.g., nodes must be added). So I would write here: `If the graph supports node addition the nodes must be added if they do not exist, otherwise SetEdge must panic.` If it is too much "must", I think using present tense is ok. On the other hand, an implementation of Graph may support setting of loop edges, so forcing them to panic might be unnecessary, we don't need them to panic but we do not/must not rely on the that support. Then we could say (and this is just speculation, not proposal): `Implementations of SetEdge may panic if the IDs of the edge nodes are equal.`